packages/builders/src/interactions/commands/chatInput/mixins/SharedChatInputCommandOptions.ts

```diff
@@ -0,0 +1,161 @@
+import { normalizeArray, type RestOrArray } from '../../../../util/normalizeArray.js';
+import { resolveBuilder } from '../../../../util/resolveBuilder.js';
+import { ChatInputCommandAttachmentOption } from '../options/attachment.js';
+import { ChatInputCommandBooleanOption } from '../options/boolean.js';
+import { ChatInputCommandChannelOption } from '../options/channel.js';
+import { ChatInputCommandIntegerOption } from '../options/integer.js';
+import { ChatInputCommandMentionableOption } from '../options/mentionable.js';
+import { ChatInputCommandNumberOption } from '../options/number.js';
+import { ChatInputCommandRoleOption } from '../options/role.js';
+import { ChatInputCommandStringOption } from '../options/string.js';
+import { ChatInputCommandUserOption } from '../options/user.js';
+import type { ApplicationCommandOptionBase } from './ApplicationCommandOptionBase.js';
+
+export interface SharedChatInputCommandOptions {
+	options?: ApplicationCommandOptionBase[];
+}
+/**
+ * This mixin holds symbols that can be shared in chat input command options.
+ *
+ * @typeParam TypeAfterAddingOptions - The type this class should return after adding an option.
+ */
+export class SharedChatInputCommandOptions {
+	protected declare readonly data: SharedChatInputCommandOptions;
+
+	/**
+	 * Adds boolean options.
+	 *
+	 * @param options - Options to add
+	 */
+	public addBooleanOptions(
```

Comment:
Seems like there's no way to clear options after adding them, and we can't really add a `setOptions()` method I think.

Maybe we should add `clearOptions()` (even though `options` are not optional)?

Reply:
I don't really see a heavy need for it here, I think it's ok w/o for now.

Reply:
Actually nevermind, I'll add a splice method here too.